### ISSUE TITLE
$scope.data doesn't seem to update when new task is pushed into. 

### ISSUE THREAD

**Issue by xskid** [angular-gantt/angular-gantt#553]
Editing is fine. Seems that, when I edit a task something is fired and gantt seems to update correctly.

When i'm trying to push new tasks, like: $scope.data[index].tasks.push(newTask);

$scope.data has the new task but gantt doesnt update properly. Which means, new task is created but the duration of the parent row doesnt.

And i'm getting this error too:

"TypeError: Cannot read property 'isBefore' of undefined at /libs/gantt/angular-gantt-plugins.js:806:52", somewhere near "angular.forEach(allTasks, function (task) {"

**Comment by xskid** [comment]
update: @ gantt.groups 

 $scope.gantt.api.tasks.on.add($scope, function (task) {
            updateTaskGroup();
        });

Seems to work now.

**Comment by xskid** [comment]
update: @ gantt.group, about the "issue 1", 

$scope.gantt.api.columns.generate(); seems to solve it. Now, when I add or remove a any task, the date range view refresh.

But I'm wondering if it is the proper way.

**Comment by Toilal** [comment]
Yes you can use this. Maybe adding new tasks to data is not well supported, let's keep this issue opened though. I don't have time to test.

**Comment by akoskm** [comment]
@Toilal just to clarify, changes to the underlying object of angular-gantt - in this case `$scope.data` - shouldn't be automatically reflected to the chart, and this is the correct behavior. Is this correct?

**Comment by Toilal** [comment]
$scope.data should be automatically reflected to the chart.

**Comment by p3nt4** [comment]
Hi, any updates on this?
I have the same issue but regenerating columns is not seamless so I cant use it. 
What's strange is that the chart updates when clicked on but not before.

**Comment by minmur** [comment]
Hello.
Had similar problem, but found out that it's not working when `data` is passed via `options` attribute. When `data` is used as `data` attribute, everything works as expected, i.e. new task appears in the chart.

```html
<!-- No go -->
<gantt options={data: [..]}`></gantt>

<!-- Good to go -->
<gantt data=[..]></gantt>
```

**Comment by Toilal** [comment]
Thanks for this info, it should help to fix
